dvc/utils/fs.py

```diff
@@ -31,3 +32,24 @@ def get_mtime_and_size(path):
     # State of files handled by dvc is stored in db as TEXT.
     # We cast results to string for later comparisons with stored values.
     return str(int(nanotime.timestamp(mtime))), str(size)
+
+
+class BasePathNotInCheckedPathException(DvcException):
+    def __init__(self, path, base_path):
+        msg = "Path: {} does not overlap with base path: {}".format(
+            path, base_path
+        )
+        super(DvcException, self).__init__(msg)
+
+
+def contains_symlink_up_to(path, base_path=""):
```

Comment:
I don't think you are using this function without base_path anywhere, so is there a reason to make it optional?